libraries/exoplayer_rtsp/src/main/java/androidx/media3/exoplayer/rtsp/RtspMediaTrack.java

```diff
@@ -311,33 +319,30 @@ private static void processAacFmtpAttribute(
   }
 
   /**
-   * Parses an MPEG-4 Audio Stream Mux configuration, as defined in ISO/IEC14496-3. FMTP attribute
-   * contains config which is a byte array containing the MPEG-4 Audio Stream Mux configuration to
-   * parse.
+   * Returns a {@link Pair} of sample rate and channel count, by parsing the
+   *  MPEG4 Audio Stream Mux configuration.
+   *
+   * <p>fmtp attribute {@code config} includes the MPEG4 Audio Stream Mux
+   * configuration (ISO/IEC14496-3, Chapter 1.7.3).
    */
-  private static Pair<Integer, Integer> getSampleRateAndChannelCountFromAudioConfig(
-      ImmutableMap<String, String> fmtpAttributes,
-      int channelCount,
-      int sampleRate) {
-    @Nullable String configInput = fmtpAttributes.get(PARAMETER_MP4V_CONFIG);
-    if (configInput != null && configInput.length() % 2 == 0) {
-      byte[] configBuffer = Util.getBytesFromHexString(configInput);
-      ParsableBitArray scratchBits = new ParsableBitArray(configBuffer);
-      int audioMuxVersion = scratchBits.readBits(1);
-      if (audioMuxVersion == 0) {
-        checkArgument(scratchBits.readBits(1) == 1, "Invalid allStreamsSameTimeFraming.");
-        scratchBits.readBits(6);
-        checkArgument(scratchBits.readBits(4) == 0, "Invalid numProgram.");
-        checkArgument(scratchBits.readBits(3) == 0, "Invalid numLayer.");
-        AacUtil.Config aacConfig = null;
-        try {
-          aacConfig = AacUtil.parseAudioSpecificConfig(scratchBits, false);
-        } catch (ParserException e) {
-          throw new IllegalArgumentException(e);
-        }
-        sampleRate = aacConfig.sampleRateHz;
-        channelCount = aacConfig.channelCount;
+  private static Pair<Integer, Integer> getSampleRateAndChannelCount(String configInput) {
+    int channelCount = 0, sampleRate = 0;
+    byte[] configBuffer = Util.getBytesFromHexString(configInput);
+    ParsableBitArray scratchBits = new ParsableBitArray(configBuffer);
+    int audioMuxVersion = scratchBits.readBits(1);
+    if (audioMuxVersion == 0) {
+      checkArgument(scratchBits.readBits(1) == 1, "Invalid allStreamsSameTimeFraming.");
+      scratchBits.readBits(6);
+      checkArgument(scratchBits.readBits(4) == 0, "Invalid numProgram.");
```

Comment:
change error message to "Only supports one program", ditto for layer, and `allStreamsSameTimeFraming`.